ConfigApp/EffectConfig_incompatibility.xaml.cs

```diff
@@ -0,0 +1,172 @@
+            foreach (EffectType item in (EffectType[])Enum.GetValues(typeof(EffectType)))
+            {
+                if (item == EffectType._EFFECT_ENUM_MAX || EffectsMap[item].Id == effectInfo.Id)
+                {
+                    continue;
+                }
+                ListData listData = new ListData();
+                listData.name = EffectsMap[item].Name;
+                if (effectData.IncompatibleWith != null)
+                {
+                    if (effectData.IncompatibleWith.Contains(item))
+                    {
+                        listData.inList = true;
+                    }
+                }
+                effects.Add(item, listData);
+            }
+            
+            foreach (KeyValuePair<EffectType, ListData> item in effects)
+            {
+                string name = item.Value.name;
+                bool inList = item.Value.inList;
+                if (inList)
+                {
+                    effectconf_custom_incompatibility_remove_list.Items.Add(name);
+                } 
+                else
+                {
+                    effectconf_custom_incompatibility_add_list.Items.Add(name);
+                }
+            }
+            SortLists();
+        }
+
+        public EffectConfig_incompatibility(EffectData effectData, EffectInfo effectInfo)
+        {
+            InitializeComponent();
+
+            if (String.IsNullOrEmpty(effectData.CustomName))
+            {
+                Title = effectInfo.Name + " - Incompatibility Config";
+
+            }
+            else
+            {
+                Title = effectData.CustomName + " - Incompatibility Config";
+            }
+
+            InitList(effectData, effectInfo);
+
+            effectconf_custom_incompatibility_remove_button.Click += (object sender, RoutedEventArgs e) =>
+            {
+                string selectedName = (string)effectconf_custom_incompatibility_remove_list.SelectedItem;
+                if (String.IsNullOrEmpty(selectedName)) { return; }
+                EffectType selectedType = EffectType._EFFECT_ENUM_MAX;
+                foreach (KeyValuePair<EffectType, ListData> item in effects)
+                {
+                    if (item.Value.name == selectedName)
+                    {
+                        selectedType = item.Key;
+                        break;
+                    }
+                }
+                if (selectedType == EffectType._EFFECT_ENUM_MAX) { return; }
+                ListData selectedData = effects[selectedType];
+                selectedData.inList = false;
+                effects[selectedType] = selectedData;
+                effectconf_custom_incompatibility_remove_list.Items.Remove(selectedName);
+                effectconf_custom_incompatibility_add_list.Items.Add(selectedName);
+                SortLists();
+            };
+
+            effectconf_custom_incompatibility_add_button.Click += (object sender, RoutedEventArgs e) =>

```

Comment:
Extracting those lambdas into separate functions help the readability of this code

Reply:
The reason I made them lambdas is because I needed to access `effectData` and `effectInfo` variables. Not the best at c#. Is there a way to access them in a separate function?